### DISCUSSION TITLE
Browser Exam Key in SEB Win v3.3.1

### DISCUSSION THREAD

**Discussion by nvarghese-rm** [SafeExamBrowser/seb-win-refactoring#348] [General]
Hi 

We  referred to https://safeexambrowser.org/developer/documents/SEB-Specification-BrowserExamKey.pdf when developing the browser exam key check in our test application. Since upgrading to SEB Windows v3.0, the browser key implementation has stopped working.

I was wondering if there has been a change with the way the Browser Exam Keys are generated in v3.0 compared to v2.3 that we need to be aware of?

**Comment by dbuechel** [comment]
Yes, I think that document is out of date as it describes the algorithm for version 2.x. In 3.x, we adopted the algorithm, specifically regarding step 1 "Generate code signature of of the whole SEB application (binaries and resources)", see https://github.com/SafeExamBrowser/seb-win-refactoring/blob/master/SafeExamBrowser.Configuration/Cryptography/KeyGenerator.cs#L36.

**Reply by danschlet** [reply]
Please note that the Browser Exam Key isn't intended to be calculated server-side. We only support it to be copy-pasted from an SEB client configuration interface to a web application.
The [Config Key](https://safeexambrowser.org/developer/seb-config-key.html) is designed to be server-generated.
